inginious/frontend/plugins/notebooks_grader/api/notebook_grader_api.py

```diff
@@ -0,0 +1,185 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of UNCode. See the LICENSE and the COPYRIGHTS files for
+# more information about the licensing of this file.
+"""Notebooks grader serverless module"""
+
+from hashlib import sha512
+import datetime
+import web
+import inginious.frontend.pages.api._api_page as api
+from inginious.frontend.plugins.utils import get_mandatory_parameter
+from inginious.common.course_factory import CourseNotFoundException
+
+class NotebookGradingAPI(api.APIAuthenticatedPage):
+    """API definition for get and set grader of a test"""
+
+    def API_GET(self): # pylint: disable=arguments-differ
+        """GET: API get grader from a test of a task
+            params: course_id
+                    task_id
+                    test_id
+            returns: 200 and task grader, functions names to evaluate and variables names to evaluate
+        """
+        request_params = web.input()
+
+        course_id = get_mandatory_parameter(request_params, "course_id")
+        task_id = get_mandatory_parameter(request_params, "task_id")
+        test_id = get_mandatory_parameter(request_params, "test_id")
+
+        try:
+            course = self.course_factory.get_course(course_id)
+        except CourseNotFoundException as course_not_found:
+            raise api.APINotFound("Course not found") from course_not_found
+
+        course_staff = course.get_staff()
+        course_students = self.user_manager.get_course_registered_users(
+            course, with_admins=False)
+
+        username = self.user_manager.session_username()
+
+        if username in course_students and username not in course_staff:
```

Comment:
Staff users can't retrieve the grader from the server, please enable that students, tutors and admins of a course could get it in order to verify funcionality

Reply:
Remember that **only students** can use the `GET` endpoint in order to get grader. Here, staff are admins and tutors can't get grader of a test (due to use case).
To verify functionality, staff can run graders locally before send those to the server

Reply:
Sure but without the change, staff can't make submissions because they need to get the grader from thde server, the same case that in the other comment.

Reply:
Please review use case, staff never uses this endpoint

Reply:
The same below